Lib/test/test_dis.py

```diff
@@ -219,6 +219,45 @@ def bug42562():
           8 RETURN_VALUE
 """
 
+bug46724 = compile("while not (a < b < c):\n    pass", "", "exec")
```

Comment:
You will need to build the bytecode explicitly, as this depends on the behavior of the compiler.

Reply:
Sure, I can do that.

Many of the other tests seem to depend on compiler behavior in this file, by comparing the code object from a function with a string of the `dis` output. But is the preference for new tests to not depend on the compiler?

Reply:
Note that Mark also opened #31326 which changes this construct to no longer produce a negative offset.

Reply:
Does that mean this PR should be closed? Is providing a negative oparg "supposed" to happen with python bytecode?

Reply:
This does need fixing, so this PR is useful.

Given the sequence of bytecodes that produces this dis output:
```
     JUMP_FORWARD         4294967280 to 8589934616
```
this PR does the right thing. This is the correct output:
```
     JUMP_FORWARD        -16 (to 24)
```

Reply:
> Is providing a negative oparg "supposed" to happen with python bytecode?

No, but `dis` should be able to cope with it.